src/wallet/wallet.h

```diff
@@ -147,6 +147,28 @@ class CKeyPool
     }
 };
 
+/** Record info about last kernel stake operation (time and chainTip)**/
+class CStakerStatus {
+private:
+    const CBlockIndex* tipLastStakeAttempt;
+    int64_t timeLastStakeAttempt;
+public:
+    const CBlockIndex* GetLastTip() const { return tipLastStakeAttempt; }
+    uint256 GetLastHash() const
+    {
+        return (tipLastStakeAttempt == nullptr ? 0 : tipLastStakeAttempt->GetBlockHash());
+    }
+    int64_t GetLastTime() const { return timeLastStakeAttempt; }
+    void SetLastTip(const CBlockIndex* lastTip) { tipLastStakeAttempt = lastTip; }
+    void SetLastTime(const uint64_t lastTime) { timeLastStakeAttempt = lastTime; }
+    void Update(CBlockIndex* lastTip, const uint64_t lastTime)
+    {
+        SetLastTip(lastTip);
+        SetLastTime(lastTime);
+    }
```

Comment:
Is it me or this `Update` method is never called? You seems to be only using the two setters.

Reply:
It is indeed never called.
It was included only to make the class "complete" with a general setter.
But we can of course remove it for now, and include it only if/when it'll be required.